From 2276f5cd4e3db131be3f5e5c2ecde425a3389bdd Mon Sep 17 00:00:00 2001
From: Jinsun Yoo <jinsunyoo332@gmail.com>
Date: Tue, 12 Mar 2024 23:19:56 -0400
Subject: [PATCH] Release constraint for COMM_RECV nodes

---
 astra-sim/workload/HardwareResource.cc | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/astra-sim/workload/HardwareResource.cc b/astra-sim/workload/HardwareResource.cc
index b631fcf..1a9ae3b 100644
--- a/astra-sim/workload/HardwareResource.cc
+++ b/astra-sim/workload/HardwareResource.cc
@@ -26,9 +26,12 @@ void HardwareResource::occupy(const shared_ptr<Chakra::ETFeederNode> node) {
       assert(num_in_flight_gpu_comp_ops == 0);
       ++num_in_flight_gpu_comp_ops;
     } else{
+      if (node->type() == ChakraNodeType::COMM_RECV_NODE) {
+        return; 
+      }
       assert(num_in_flight_gpu_comm_ops == 0);
       ++num_in_flight_gpu_comm_ops;
-    }
+      }
   }
 }
 
@@ -41,6 +44,9 @@ void HardwareResource::release(const shared_ptr<Chakra::ETFeederNode> node) {
       --num_in_flight_gpu_comp_ops;
       assert(num_in_flight_gpu_comp_ops == 0);
     } else {
+      if (node->type() == ChakraNodeType::COMM_RECV_NODE) {
+        return;
+      }
       --num_in_flight_gpu_comm_ops;
       assert(num_in_flight_gpu_comm_ops == 0);
     }
@@ -63,6 +69,9 @@ bool HardwareResource::is_available(
           return false;
         }
       } else {
+        if (node->type() == ChakraNodeType::COMM_RECV_NODE){
+          return true;
+        }
         if (num_in_flight_gpu_comm_ops == 0) {
           return true;
         } else {
-- 
2.25.1

